### PR TITLE
SRCH-3668 Integrate DAP Google Analytics

### PR DESCRIPTION
**Pull request by krbhavith**
## Summary
- Add DAP GA to SERP redesign results page and remove rendering custom GA
- Add GTM for SERP redesign pages.
- Add third party tracking in SERP redesign pages.
 
### Checklist
Please ensure you have addressed all concerns below before marking a PR "ready for review" or before requesting a re-review. If you cannot complete an item below, replace the checkbox with the ⚠️ `:warning:` emoji and explain why the step was not completed.
 
#### Functionality Checks
 
- [x] You have merged the latest changes from the target branch (usually `main`) into your branch.
  
- [x] Your primary commit message is of the format **SRCH-#### \<description\>** matching the associated Jira ticket.

- [x] PR title is either of the format **SRCH-#### \<description\>** matching the associated Jira ticket (i.e. "SRCH-123 implement feature X"), or **Release - SRCH-####, SRCH-####, SRCH-####** matching the Jira ticket numbers in the release.
 
- [x] Automated checks pass. If Code Climate checks do not pass, explain reason for failures:
 
#### Process Checks

- [x] You have specified at least one "Reviewer".

### CODE REVIEW

File: app/views/layouts/searches_redesign.html.haml [hunk 1]
```diff
@@ -1,6 +1,7 @@
 !!!strict
 %html
   %head
+    = render "/shared/google_tag_manager_script"
```
[the remark this answers is not in the document]
Comment (ddzz) [reply app/views/layouts/searches_redesign.html.haml:4]: This seems like the correct spot for the redesign.

File: app/views/layouts/searches_redesign.html.haml [hunk 2]
```diff
@@ -14,3 +15,5 @@
     = stylesheet_link_tag 'searches_redesign'
   %body
     #main-content= yield
+    = render "/shared/google_tag_manager_noscript"
+    = render "/shared/affiliate_google_analytics"
```
Comment (ddzz) [inline app/views/layouts/searches_redesign.html.haml:19]: Should we move this outside of the `<body>`?

Comment (dawnpm) [reply app/views/layouts/searches_redesign.html.haml:19]: thanks @ddzz - historically we've had it in the body, but looking at https://github.com/digital-analytics-program/gov-wide-code/blob/master/documentation/GSA%20DAP%204.1%20-%20Quick%20Guide.pdf, they recommend it goes in the head. sorry @krbhavith !

File: app/views/layouts/searches_redesign.html.haml [hunk 3]
```diff
@@ -14,3 +15,5 @@
     = stylesheet_link_tag 'searches_redesign'
   %body
     #main-content= yield
+    = render "/shared/google_tag_manager_noscript"
+    = render "/shared/analytics"
```
Comment (ddzz) [inline app/views/layouts/searches_redesign.html.haml:19]: Can you move these to the `<head>`?

Comment (krbhavith) [reply app/views/layouts/searches_redesign.html.haml:19]: Generally it's used in the body tag right?

Comment (krbhavith) [reply app/views/layouts/searches_redesign.html.haml]: Moved google analytics script to `<head>` and added external tracking code at the end of `<body>` tag.

**Review by ddzz — DISMISSED** [review]
LGTM

**Comment by krbhavith** [comment]
@ddzz updated comments around GTM and GA script as requested by Dawn, please re-review

**Pull request merged**